components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -0,0 +1,241 @@
+
+    @Mock
+    ServiceProvider serviceProvider;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @Test
+    public void testBuildIDToken() throws Exception {
+        request1 = mock(OAuthTokenReqMessageContext.class);
+        tokenReqDTO = mock(OAuth2AccessTokenReqDTO.class);
+        when(request1.getOauth2AccessTokenReqDTO()).thenReturn(tokenReqDTO);
+        when(tokenReqDTO.getTenantDomain()).thenReturn("tenant1");
+        identityProvider = mock(IdentityProvider.class);
+        mockStatic(IdentityProviderManager.class);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+        when(identityProviderManager.getResidentIdP(anyString())).thenReturn(identityProvider);
+
+        FederatedAuthenticatorConfig[] federatedAuthenticatorConfigs = new FederatedAuthenticatorConfig[10];
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(federatedAuthenticatorConfigs);
+
+        mockStatic(IdentityApplicationManagementUtil.class);
+        when(IdentityApplicationManagementUtil.getFederatedAuthenticator(federatedAuthenticatorConfigs, "tenant2")).thenReturn(federatedAuthenticatorConfig);
+        Property[] properties = new Property[1];
+        Property property = new Property();
+        property.setName("IdPEntityId");
+        property.setValue("localhost");
+        properties[0] = property;
+
+        federatedAuthenticatorConfig.setProperties(properties);
+        FederatedAuthenticatorConfig[] federatedAuthenticatorConfig1 = new FederatedAuthenticatorConfig[10];
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(federatedAuthenticatorConfig1);
+
+        mockStatic(IdentityApplicationManagementUtil.class);
+        when(IdentityApplicationManagementUtil.getFederatedAuthenticator(federatedAuthenticatorConfigs, IdentityApplicationConstants.Authenticator.OIDC.NAME)).thenReturn(federatedAuthenticatorConfig);
+        when(IdentityApplicationManagementUtil.getProperty(any(Property[].class), anyString())).thenCallRealMethod();
+
+        when(federatedAuthenticatorConfig.getProperties()).thenReturn(properties);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 3);
+        mockStatic(OAuth2Util.class);
+        when(oAuthServerConfiguration.getOpenIDConnectIDTokenExpiration()).thenReturn(String.valueOf((int) 78383));
+        when(OAuth2Util.mapDigestAlgorithm(any(JWSAlgorithm.class))).thenReturn(String.valueOf(JWSAlgorithm.RS256));
+
+        when(request1.getAuthorizedUser()).thenReturn(authenticatedUser);
+        when(authenticatedUser.getAuthenticatedSubjectIdentifier()).thenReturn("Subject_Identifier");
```

Comment:
Let's use a private String constant for this like SUBJECT_IDENTIFIER = "Subject_Identifier"